vsphere/resource_vsphere_virtual_machine_test.go

```diff
@@ -1,1726 +1,1548 @@
+// and that the appropriate gateway is present.
+//
+// This uses VMware tools to check this, so it needs to be installed on the
+// guest.
+func testAccResourceVSphereVirtualMachineCheckNet(expectedAddr, expectedPrefix, expectedGW string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		res, err := strconv.Atoi(expectedPrefix)
+		if err != nil {
+			return fmt.Errorf("bad: %s", err)
+		}
+		expectedPrefixInt := int32(res)
+
+		props, err := testGetVirtualMachineProperties(s, "vm")
+		if err != nil {
+			return err
+		}
+		var v4gw, v6gw net.IP
+		for _, s := range props.Guest.IpStack {
+			if s.IpRouteConfig != nil {
+				for _, r := range s.IpRouteConfig.IpRoute {
+					switch r.Network {
+					case "0.0.0.0":
+						v4gw = net.ParseIP(r.Gateway.IpAddress)
+					case "::":
+						v6gw = net.ParseIP(r.Gateway.IpAddress)
+					}
+				}
+			}
+		}
+		for _, n := range props.Guest.Net {
+			if n.IpConfig != nil {
+				for _, addr := range n.IpConfig.IpAddress {
+					ip := net.ParseIP(addr.IpAddress)
+					if !ip.Equal(net.ParseIP(expectedAddr)) {
+						continue
+					}
+					if addr.PrefixLength != expectedPrefixInt {
+						continue
+					}
+					var mask net.IPMask
+					if ip.To4() != nil {
+						mask = net.CIDRMask(int(addr.PrefixLength), 32)
+					} else {
+						mask = net.CIDRMask(int(addr.PrefixLength), 128)
+					}
+					switch {
+					case ip.Mask(mask).Equal(v4gw.Mask(mask)):
+						if net.ParseIP(expectedGW).Equal(v4gw) {
+							return nil
+						}
+					case ip.Mask(mask).Equal(v6gw.Mask(mask)):
+						if net.ParseIP(expectedGW).Equal(v6gw) {
+							return nil
+						}
+					}
+				}
+			}
+		}
+
+		return fmt.Errorf("could not find IP %s/%s, gateway %s", expectedAddr, expectedPrefix, expectedGW)
 	}
-	if v := os.Getenv("VSPHERE_DATASTORE"); v != "" {
-		datastoreOpt = fmt.Sprintf("        datastore = \"%s\"\n", v)
+}
+
+// testAccResourceVSphereVirtualMachineCheckStaticMACAddr is a check to look
+// for the MAC address defined in the static MAC address test on the first
+// network interface.
+func testAccResourceVSphereVirtualMachineCheckStaticMACAddr() resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		props, err := testGetVirtualMachineProperties(s, "vm")
+		if err != nil {
+			return err
+		}
+		actual := props.Guest.Net[0].MacAddress
+		expected := "06:5c:89:2b:a0:64"
+		if expected != actual {
+			return fmt.Errorf("expected MAC address to be %s, got %s", expected, actual)
+		}
+		return nil
+	}
+}
+
+// testAccResourceVSphereVirtualMachineCheckAnnotation is a check to ensure
+// that a VM's annotation is correctly set in the annotation test.
+func testAccResourceVSphereVirtualMachineCheckAnnotation() resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		props, err := testGetVirtualMachineProperties(s, "vm")
+		if err != nil {
+			return err
+		}
+		expected := "Managed by Terraform"
+		actual := props.Config.Annotation
```

Comment:
And constant here?

Reply:
This is the actual value which we need to pull from the VM state (`Config` here denotes the configuration section of the retrieved properties).

Reply:
My bad. Didn't notice the lack of ""